x-pack/plugins/security_solution/public/cases/components/use_create_case_modal/index.tsx

```diff
@@ -22,13 +22,18 @@ export const useCreateCaseModal = ({ onCaseCreated }: Props) => {
   const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
   const closeModal = useCallback(() => setIsModalOpen(false), []);
   const openModal = useCallback(() => setIsModalOpen(true), []);
+  const onSuccess = useCallback(
+    (theCase) => {
+      onCaseCreated(theCase);
+      closeModal();
+    },
+    [onCaseCreated, closeModal]
+  );
 
   const Modal: React.FC = useCallback(
     () =>
-      isModalOpen ? (
-        <CreateCaseModal onCloseCaseModal={closeModal} onCaseCreated={onCaseCreated} />
-      ) : null,
-    [closeModal, isModalOpen, onCaseCreated]
+      isModalOpen ? <CreateCaseModal onCloseCaseModal={closeModal} onSuccess={onSuccess} /> : null,
```

Comment:
I just saw the callback to render a component, I believe that's more a hack than a pattern. I will prefer it if you change it to that

```
const state = useMemo(
    () => ({
      modal: (
        <CreateCaseModal
          openModal={isModalOpen}
          onCloseCaseModal={closeModal}
          onSuccess={onSuccess}
        />
      ),
      isModalOpen,
      closeModal,
      openModal,
    }),
    [isModalOpen, closeModal, openModal, onSuccess]
  );
```

Reply:
Please follow the same pattern in this file -> x-pack/plugins/security_solution/public/cases/components/use_all_cases_modal/index.tsx

Reply:
The same pattern is used there. I will change them both to your proposal.